pydov/types/monster.py

```diff
@@ -7,6 +7,121 @@
 import datetime
 import numpy as np
 
+class TijdstipField(_CustomXmlField):
+    """Field for retrieving the time from the relevant XML
+    field."""
+
+    def __init__(self, name, definition, source_xpath):
+        super().__init__(
+            name=name,
+            definition=definition,
+            datatype='string',
+            notnull=False
+        )
+        self.source_xpath = source_xpath
+
+    def calculate(self, cls, tree):
+        tijdstip = cls._parse(
+            func=tree.findtext,
+            xpath=self.source_xpath + 'tijdstip',
+            namespace=None,
+            returntype='string'
+        )
+        if tijdstip is not np.nan:
+            return tijdstip
+
+        datumtijd = cls._parse(
+            func=tree.findtext,
+            xpath=self.source_xpath + 'datumtijd',
+            namespace=None,
+            returntype='string'
+        )
+        if datumtijd is not np.nan:
+            timestamp = datetime.datetime.fromisoformat(datumtijd)
+            return timestamp.strftime('%H:%M:%S')
+
+        return np.nan
+
+
+class DatumField(_CustomXmlField):
+    """Field for retrieving the date from the relevant XML
+    field."""
+
+    def __init__(self, name, definition, source_xpath):
```

Comment:
I would call this `base_xpath` instead of `source_xpath` to indicate this is not the full XPath expression.

Can you also add a docstring to this function?